create/application.go

```diff
@@ -102,10 +103,18 @@ func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
 		if err := auth.Valid(); err != nil {
 			return fmt.Errorf("the credentials are given but they are empty: %w", err)
 		}
-		// for git auth we create a separate secret and then reference it in the app.
+
 		secret := auth.Secret(newApp)
-		if err := client.Create(ctx, secret); err != nil {
-			return fmt.Errorf("unable to create git auth secret: %w", err)
+		if err := client.Get(ctx, types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}, secret); err != nil {
+			// for git auth we create a separate secret and then reference it in the app.
+			if err := client.Create(ctx, secret); err != nil {
+				return fmt.Errorf("unable to create git auth secret: %w", err)
+			}
+		} else {
+			fmt.Println("updating git auth credentials")
+			if err := client.Update(ctx, secret); err != nil {
```

Comment:
It's a bit dangerous to just update the secret and discard whatever was in there. The user could create a secret on the API with the same name, unrelated to this app and then have his data overwritten. So we should check if the secret has the `ManagedByAnnotation: nctl` and only then update it.

Reply:
added